### PR TITLE
TASK: Respect `startingPoint` in ReferenceDataLoader

### LINKED ISSUE #1496: ReferenceEditor does not respect a `startingPoint`
Given this configuration:
```
    news:
      type: reference
      ui:
        reloadIfChanged: true
        inspector:
          editorOptions:
            nodeTypes:
              - 'Yeebase.t3nDe:Article'
            startingPoint: /articles
```

The startingPoint is not respected. Therefore the ReferenceEditor won't find any results

### PR DESCRIPTION
**Pull request by johannessteu**
Fixes #1496

### CODE REVIEW

File: packages/neos-ui-editors/src/Editors/Reference/referenceDataLoader.js [hunk 1]
```diff
@@ -103,9 +104,13 @@ export default ({isMulti}) => WrappedComponent => {
         }
 
         getDataLoaderOptions() {
+            const contextNode = this.props.options.startingPoint ?
+                this.props.contextForNodeLinking.set('contextNode', this.props.options.startingPoint) :
```
Comment (kitsunet) [inline packages/neos-ui-editors/src/Editors/Reference/referenceDataLoader.js]: you are modifying props there. I don't think that should EVER happen. You might need to create a deep clone of contextForNodeLinking if starting point is set and then set it on that copy.

Comment (johannessteu) [reply packages/neos-ui-editors/src/Editors/Reference/referenceDataLoader.js]: Does it? contextForNodeLinking is a Immutable map and i tought that the set() creates a copy (https://facebook.github.io/immutable-js/ second code example)

Comment (johannessteu) [reply packages/neos-ui-editors/src/Editors/Reference/referenceDataLoader.js]: It's initalizied here: https://github.com/neos/neos-ui/blob/master/packages/neos-ui-redux-store/src/UI/NodeLinking/index.js#L24

Comment (kitsunet) [reply packages/neos-ui-editors/src/Editors/Reference/referenceDataLoader.js]: Oh, is it? In that case YAY, but IMHO the wording is awkward, not your fault though.. `set` shouldn't do `createDeepCloneWithThisPropertyChanged`

**Pull request merged**